torch/csrc/jit/script/concrete_module_type.cpp

```diff
@@ -1,8 +1,51 @@
 #include <torch/csrc/jit/script/concrete_module_type.h>
+#include <torch/csrc/jit/pybind_utils.h>
 
 namespace torch {
 namespace jit {
 namespace script {
+
+bool isSimpleConstant(const IValue& v) {
+  return v.isBool() || v.isDouble() || v.isInt() || \
+    v.isString() || v.isNone() || v.isDevice();
+}
+
+IValue getValidConstantFromTuple(const std::string& attr_name,
+                                 const std::vector<IValue>& l) {
+  std::vector<IValue> valid_list;
+  std::transform(l.begin(), l.end(), std::back_inserter(valid_list),
+                 [attr_name](const IValue& v) -> IValue {
+                   return getValidConstant(attr_name, v);
+                 });
+  return c10::ivalue::Tuple::create(valid_list);
```

Comment:
This logic is very weird -- it turns lists into tuples? Why is this happening at this point? This seems like a way of grandfathering in python conversion behavior into IValue conversions.  I feel like it would be more appropriate for lists to be converted to tuples in python before it gets to this stage.

Reply:
Removed. I'll keep this logic in python